app/controllers/reports/regions_controller.rb

```diff
@@ -11,11 +11,23 @@ class Reports::RegionsController < AdminController
 
   def index
     authorize { current_admin.accessible_facilities(:view_reports).any? }
-    @organizations = current_admin.accessible_facilities(:view_reports)
-      .flat_map(&:organization)
-      .uniq
-      .compact
-      .sort_by(&:name)
+
+    if current_admin.feature_enabled?(:regions_reports)
+      @organizations = current_admin.accessible_facilities(:view_reports)
+        .flat_map(&:organization)
+        .uniq
+        .compact
+        .sort_by(&:name)
+    else
+      @accessible_regions = current_admin.accessible_facility_regions(:view_reports).each_with_object({}) { |facility, result|
```

Comment:
Here we grab the lowest level accessible node for the current admin, a facility, and then build the tree back up from that to be able to show the correct hierarchy in the view.